rundk/main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/spf13/pflag"
+	"knative.dev/test-infra/pkg/helpers"
+	"knative.dev/test-infra/pkg/interactive"
+)
+
+var (
+	image                     string
+	entrypoint                string
+	dindEnabled               bool
+	useLocalGcloudCredentials bool
+	useLocalKubeconfig        bool
+	mounts                    []string
+	mandatoryEnvVars          []string
+	optionalEnvVars           []string
+)
+
+func main() {
+	pflag.StringVar(&image, "test-image", "gcr.io/knative-tests/test-infra/prow-tests:stable", "The image we use to run the test flow.")
+	pflag.StringVar(&entrypoint, "entrypoint", "runner.sh", "The entrypoint executable that runs the test commands.")
+	pflag.BoolVar(&dindEnabled, "enable-docker-in-docker", false, "Enable running docker commands in the test flow. "+
+		"By enabling this the container will share the same docker daemon in the host machine, so be careful when using it.")
+	pflag.BoolVar(&useLocalGcloudCredentials, "use-local-gcloud-credentials", false, "Use the same gcloud credentials as local, which can be set "+
+		"either by setting env var GOOGLE_CLOUD_APPLICATION_CREDENTIALS or from ~/.config/gcloud")
+	pflag.BoolVar(&useLocalKubeconfig, "use-local-kubeconfig", false, "Use the same kubeconfig as local, which can be set "+
+		"either by setting env var KUBECONFIG or from ~/.kube/config")
+	pflag.StringSliceVar(&mounts, "mounts", []string{}, "A list of extra folders or files separated by comma that need to be mounted to run the test flow."+
+		"It must be in the format of `source1:target1,source2:target2,source3:target3`.")
+	pflag.StringSliceVar(&mandatoryEnvVars, "mandatory-env-vars", []string{}, "A list of env vars separated by comma that must be set on local.")
+	pflag.StringSliceVar(&optionalEnvVars, "optional-env-vars", []string{}, "A list of env vars separated by comma that optionally need to be set on local.")
+	pflag.Parse()
+
+	cmd, cancel := setup()
+	defer cancel()
+
+	run(cmd, pflag.Args()...)
+}
+
+func setup() (interactive.Docker, func()) {
+	var err error
+
+	// Until everyone is using 20.xx version of docker (see https://github.com/docker/cli/pull/1498) adding the --pull flag,
+	//  need to separately pull the image first to be sure we have the latest
+	pull := interactive.NewCommand("docker", "pull", image)
+	if err = pull.Run(); err != nil {
+		log.Fatal("Error pulling the test image: ", err)
+	}
+
+	builtUpDefers := make([]func(), 0)
+
+	// Setup command
+	cmd := interactive.NewDocker()
+	envs := interactive.Env{}
+
+	// Setup temporary directory to save the artifacts
+	tmpDir, err := ioutil.TempDir("/tmp", "prow-docker.")
```

Comment:
Is the template `prow-docker.` working as expected? From the documentation [1], it seems `*` should be used to avoid overwriting.

[1] https://golang.org/pkg/io/ioutil/#TempDir

Reply:
The doc also mentioned `The directory name is generated by taking pattern and applying a random string to the end.`, so if it does not have `*`, it will simply do appending instead of replacing so we will get `prow-docker.xyz`. I have verified it on my local.